lookout/style/typos/research/eval_dataset/filter_dataset.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import pandas
+import spacy
+
+from sourced.ml.algorithms import TokenParser
+
+
+def filter_test_dataset(dataset: str, filtered_dataset: str) -> None:
+    """
+    Filter non-typo-ed identifiers from the dataset.
```

Comment:
```suggestion
    Extract non-typo-ed identifiers from the dataset.
```
Someone has to write a linter rule to disable repeating the function name parts in the docstring.

Reply:
Don't you think that it's cool if the function name really reflects the best way you would describe it's work in a text?
Anyway I thought of the better name which gives clearer understanding of what the function does.

Reply:
Totally cool, that's right. Not cool to have a docstring which does not give any additional information about the function, that's my point :)

Reply:
I particularly don't like the word "filter" in source code because it can mean different things:

1. Remove something and return the rest.
2. Extract something, return it, and discard the rest.

Those are two opposite actions. Different programmers all over the world keep using "filter" at random and it really puzzles every time when you read their code. This is why repeating "filter" in the docstring is so unhelpful.

Reply:
Totally agree, I think I myself sometimes use it for both, and always feel confused, so i really like the new name:)